src/SQLStore/SimpleListLookup.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace SMW\SQLStore;
+
+/**
+ * A basic interface for fetching a simple list either from a DB or cache
+ * instance
+ *
+ * @license GNU GPL v2+
+ * @since 2.2
+ *
+ * @author mwjames
+ */
+interface SimpleListLookup {
```

Comment:
> What makes it so that users of this thing inherently need to be able to interact with the cache stuff?

No, the way `SimpleListLookup` is specified s(he) never has to know whether it is coming from a cache or not that's why `SimpleListLookup` contains all necessary methods so a consumer such as [0] only knows it is `SimpleListLookup`.

[0] https://github.com/SemanticMediaWiki/SemanticMediaWiki/commit/311dba9f34b2f080282095d1bac0dd8185edad91#diff-af75f55a8f874a95bcdbcdd5d07da2c7L52